Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6974,14 +6974,14 @@ WEBCORE_EXPORT void HTMLMediaElement::setVideoFullscreenStandby(bool value)
         });
 }
 
-void HTMLMediaElement::willBecomeFullscreenElement()
+void HTMLMediaElement::willBecomeFullscreenElement(VideoFullscreenMode mode)
 {
 #if PLATFORM(MAC) && ENABLE(VIDEO_PRESENTATION_MODE)
     HTMLMediaElementEnums::VideoFullscreenMode oldVideoFullscreenMode = m_videoFullscreenMode;
 #endif
 
-    if (m_videoFullscreenMode != VideoFullscreenModeStandard)
-        setFullscreenMode(VideoFullscreenModeStandard);
+    if (!(m_videoFullscreenMode == VideoFullscreenModeStandard || m_videoFullscreenMode == VideoFullscreenModeInWindow))
```

Comment:
since you’re doing `== standard || == inWindow` a lot, maybe it could be refactored into a static method that’s like `isStandardOrInWindow` or something like that?

Reply:
This still seems like a good idea.